ServantPersistent/src/DBTypes.hs

```diff
@@ -0,0 +1,63 @@
+{-# LANGUAGE DeriveGeneric   #-}
+module DBTypes where
+
+import Data.Aeson
+import Data.Aeson.TH
+import Data.Text
+import GHC.Generics
+import Control.Lens
+import Types
+import Models
+
+type TenantID = Key DBTenant
+type Tenant = DBTenant
+type TenantOutput = DBTenant
+type UserID = Key DBUser
+
+data DBError = TenantNotFound TenantID
+             | UserNotFound UserID deriving (Eq, Show)
+data UserCreationError = UserExists Text
+                       | TenantDoesn'tExist Text
+                        deriving (Eq, Show)
+
+data Role = SiteAdmin
```

Comment:
Does the [roles list screen](https://vacationlabs.github.io/haskell-webapps/ui-mockups/roles.html) and [edit role screen](https://vacationlabs.github.io/haskell-webapps/ui-mockups/role-edit.html) make this clearer?